### ISSUE TITLE
[LINT_BUG]: box_unused_att_pkg_fun_linter() doesn't recognize replacement functions

### ISSUE THREAD

**Issue by calderonsamuel** [Appsilon/box.linters#181]
### box.linters version

box.linters: ‘0.10.5’
R 4.5.1
Windows 11

### Sample source code to lint

``` r
code1 <- "box::use(
  methods[
    `slot<-`,
    setClass,
  ],
)

setClass('Person', slots = c(name = 'character'))
person <- new('Person')
slot(person, 'name') <- 'John' # using replacement function here
"

lintr::lint(code1, linters = box.linters::box_unused_att_pkg_fun_linter())
#> <text>:3:5: warning: [box.linters::box_unused_att_pkg_fun_linter] Imported function unused.
#>     `slot<-`,
#>     ^~~~~~~~
```

<sup>Created on 2025-10-21 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### Lint command used

lintr::lint(code1, linters = box.linters::box_unused_att_pkg_fun_linter())

### Lint result

#> <text>:3:5: warning: [box.linters::box_unused_att_pkg_fun_linter] Imported function unused.
#>     `slot<-`,
#>     ^~~~~~~~

### Expected result

No lint results
